### PR TITLE
refactor: Use SDS Dialog and Menu

### PR DESCRIPTION
**Pull request by tihuan**
### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
1. Use SDS Dialog, Menu, and MenuItem for AuthButton
2. Clicking backdrop doesn't close modal anymore

PTAL thank you!

Screen Shot 2022-03-03 at 3 06 41 PM

demo

### CODE REVIEW

File: frontend/src/components/Header/components/AuthButtons/index.tsx [hunk 1]
```diff
@@ -41,29 +37,50 @@ const AuthButtons = (): JSX.Element | null => {
...
+        <MenuItem disabled data-testid="user-email">
+          Logged in as: {email}
+        </MenuItem>
         {curatorAPIFeature && (
-          <>
+          <div>
             <CuratorAPIKeyGenerator />
             <MenuDivider />
-          </>
+          </div>
```
Comment (seve) [inline frontend/src/components/Header/components/AuthButtons/index.tsx:79]: why the div?

**Review by seve — APPROVED** [review]

File: frontend/src/components/Header/components/AuthButtons/index.tsx (hunk 1 above)
Comment (tihuan) [reply frontend/src/components/Header/components/AuthButtons/index.tsx:79]: MUI Menu warns that it doesn't support fragments! Not sure why such limitation though

**Comment by tihuan** [comment]
Thanks for the Figma! Will update AuthButton now 🚀

**Comment by tihuan** [comment]
Updated AuthButton per [Figma](https://www.figma.com/file/3OBNVJ2Wn6aF4MdNnhVhnt/Curation-API?node-id=5%3A118) 🚀 !

Screen Shot 2022-03-03 at 4 02 30 PM

**Pull request merged**